maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/PluginOverviewRenderer.java

```diff
@@ -0,0 +1,392 @@
+            PluginDescriptor pluginDescriptor,
+            boolean hasExtensionsToLoad) {
+        super(sink, locale, i18n, project);
+
+        this.requirementsHistories = requirementsHistories;
+
+        this.pluginDescriptor = pluginDescriptor;
+
+        this.hasExtensionsToLoad = hasExtensionsToLoad;
+    }
+
+    @Override
+    public String getTitle() {
+        return getI18nString("title");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void renderBody() {
+        startSection(getTitle());
+
+        if (!(pluginDescriptor.getMojos() != null && pluginDescriptor.getMojos().size() > 0)) {
+            paragraph(getI18nString("goals.nogoal"));
+            endSection();
+            return;
+        }
+
+        paragraph(getI18nString("goals.intro"));
+
+        boolean hasMavenReport = false;
+        for (MojoDescriptor mojo : pluginDescriptor.getMojos()) {
+            if (PluginUtils.isMavenReport(mojo.getImplementation(), project)) {
+                hasMavenReport = true;
+            }
+        }
+
+        startTable();
+
+        String goalColumnName = getI18nString("goals.column.goal");
+        String isMavenReport = getI18nString("goals.column.isMavenReport");
+        String descriptionColumnName = getI18nString("goals.column.description");
+        if (hasMavenReport) {
+            tableHeader(new String[] {goalColumnName, isMavenReport, descriptionColumnName});
+        } else {
+            tableHeader(new String[] {goalColumnName, descriptionColumnName});
+        }
+
+        List<MojoDescriptor> mojos = new ArrayList<>();
+        mojos.addAll(pluginDescriptor.getMojos());
+        PluginUtils.sortMojos(mojos);
+        for (MojoDescriptor mojo : mojos) {
+            String goalName = mojo.getFullGoalName();
+
+            /*
+             * Added ./ to define a relative path
+             * @see AbstractMavenReportRenderer#getValidHref(java.lang.String)
+             */
+            String goalDocumentationLink = "./" + mojo.getGoal() + "-mojo.html";
+
+            String description;
+            if (StringUtils.isNotEmpty(mojo.getDeprecated())) {
+                description = "<strong>" + getI18nString("goal.deprecated") + "</strong> " + mojo.getDeprecated();
+            } else if (StringUtils.isNotEmpty(mojo.getDescription())) {
+                description = mojo.getDescription();
+            } else {
+                description = getI18nString("goal.nodescription");
+            }
+
+            sink.tableRow();
+            tableCell(createLinkPatternedText(goalName, goalDocumentationLink));
+            if (hasMavenReport) {
+                if (PluginUtils.isMavenReport(mojo.getImplementation(), project)) {
+                    tableCell(getI18nString("isReport"));
+                } else {
+                    tableCell(getI18nString("isNotReport"));
+                }
+            }
+            tableCell(description, true);
```

Comment:
Description handling should be moved here since it appear first here.

Reply:
see https://github.com/apache/maven-plugin-tools/pull/225#discussion_r1335221503

Reply:
Agreed. Please make a note.